### DISCUSSION TITLE
Benefit of this surface seeding over HSVS GMWMI

### DISCUSSION THREAD

**Discussion by smeisler** [dmitrishastin/SAF#1] [Q&A] [answered]
Hi,

I was curious if you had any thoughts about the benefit of the modified MRTrix surface seeding method you describe here as compared to seeding from the hybrid surface-volume segmentation (HSVS) GMWMI, which is natively in MRTrix. Do you think the benefit of the modified method is more apparent for shorter range fibers as compared to longer fibers? 

Thanks,
Steven

**Comment by dmitrishastin** [comment, marked as answer]
Hi Steven,

Thanks for taking the interest!

The way we approached short-range fibre tractography, none of the subcortical structures were used so I have not directly experimented with HSVS and I looked at freesurfer algorithm for 5tt image generation instead. I assume the two are similar in what concerns seed placement at the grey-white interface but I do not know if that is accurate. 

First of all, what prompted the introduction of the modified method: (1) the native method seemed to place seeds around and not strictly on the surface (which may well be desired depending on the effect sought), and (2) the seeds when placed in great numbers (millions) also seemed to have a slight voxelised appearance (viewed in 3D). Using mesh coordinates gets rid of these effects but, in the method's present form, the number of unique seeding coordinates is fixed. When using deterministic tractography this probably does not make sense at all, but it may be justified with probabilistic (as used in the paper) where each coordinate gives rise to multiple streamlines with different trajectories. 

Next, "the benefit" can be assessed in different ways. I am not sure what you're after - the answer may depend on that! 

Informally, my impression was that the two methods produced comparable results (at least in the context of short range tractography) and I mainly opened a pull request with MRtrix because: (1) the question of injecting coordinates for seeding has been asked before, (2) to allow others reproduce what we have done in the paper. I may or may not be using the modified method myself (in its present form) in the next work.

Formally, one of the things we looked at was the percentage of surface covered by streamlines. I have not done a strict quantitative comparison because, as briefly mentioned in the paper, it was a little difficult to reproduce the native gmwmi seeding using the experimental paradigm we were interested in (namely, performing unconstrained tractography then applying surface-based filters). When using `-seed_rejection` with gmwmi from 5tt, surface seeding seemed to do marginally better for the percentage of surface mesh vertices "covered" by streamlines - but I want to be very clear that `-seed_rejection` with gmwmi is not the same as native act-based seeding. 

I hope this answers your question. 

I am interested in developing a few methodological variations and testing them more thoroughly but as it stands I currently do not have time for this. Perhaps next year!

Best wishes,

Dmitri

**Reply by smeisler** [reply]
Great answer, thanks!

**Reply by dmitrishastin** [reply]
**An update:**

Actually there is a difference between `freesurfer` and `hsvs` algorithms at the grey-white interface. The latter uses partial volume fraction representation in 5tt whereas the former just relies on binary masks. I think that's where the apparent discretisation error of `freesurfer` algorithm was coming from. So I inspected the `hsvs`-derived seeds from the 'minimally pre-processed' HCP data (5M seeds, exactly).

Preambule: I wasn't happy with the provided intermodality registration of the HCP (1200 young adults) dataset and re-registered the data myself. Affine registration did not seem to do the best job around the grey-white interface and I adopted an ad-hoc registration sequence that relied on non-affine registration among other things. Of course, the formal MRtrix advice is to use rigid registration and simply reorient the 5tt image without regridding it (as the latter necessitates interpolation). However, for the above-mentioned reasons that did not seem to be a good option here. I did not feel that bringing all the different FreeSurfer files that `hsvs` uses into DWI space was worthwhile either. I ended up running 5ttgen with `hsvs` in anatomical space then transforming output into DWI space using (tri-)linear interpolation (with ANTs). Subsequently I made sure the different 5tt volumes added up to 1, and ran my final output through `5ttcheck` for completeness. So, using non-affine registration and interpolation so not ideal - but optimised to the best of my ability. I should mention that the reference DWI image was in the same spatial resolution as the original FreeSurfer data (1 mm isotropic).

Result: the discretisation error I talked about is no longer present. The seeds appear to be distributed in a more sheet-like pattern. There is, however, some misalignment with the mesh (also in DWI space, exactly the same registration used) as demonstrated in the image below. Whether this is a consequence of my non-affine registration, or a part of the `hsvs` algorithm, I don't know. 
 
![hsvs_seeds_DWI_small](https://user-images.githubusercontent.com/54913410/187760783-a279197b-fdb3-41ab-b148-c4d8c4dd2352.png)
